instance/main.tf

```diff
@@ -1,72 +1,104 @@
+terraform {
+  required_version = ">= 0.12"
+}
+
 module "is_ebs_optimised" {
   source        = "../is_ebs_optimised"
-  instance_type = "${var.instance_type}"
+  instance_type = var.instance_type
 }
 
 resource "aws_instance" "instance" {
-  count                       = "${var.ebs_enabled ? var.instance_count : 0}"
-  ami                         = "${var.ami}"
-  instance_type               = "${var.instance_type}"
-  key_name                    = "${var.key_name}"
-  iam_instance_profile        = "${aws_iam_instance_profile.profile.id}"
-  vpc_security_group_ids      = ["${var.sgs}"]
-  subnet_id                   = "${element(var.subnets, count.index)}"
-  disable_api_termination     = "${var.termination_protection}"
-  ebs_optimized               = "${module.is_ebs_optimised.is_ebs_optimised}"
-  associate_public_ip_address = "${var.public_ip}"
-  user_data                   = "${element(var.user_data, count.index)}"
+  count                       = var.ebs_enabled ? var.instance_count : 0
+  ami                         = var.ami
+  instance_type               = var.instance_type
+  key_name                    = var.key_name
+  iam_instance_profile        = aws_iam_instance_profile.profile[0].id
+  vpc_security_group_ids      = var.sgs
+  subnet_id                   = element(var.subnets, count.index)
+  disable_api_termination     = var.termination_protection
+  ebs_optimized               = module.is_ebs_optimised.is_ebs_optimised
+  associate_public_ip_address = var.public_ip
+  user_data                   = element(var.user_data, count.index)
 
   root_block_device {
-    volume_type           = "${var.root_vl_type}"
-    volume_size           = "${var.root_vl_size}"
-    delete_on_termination = "${var.root_vl_delete}"
+    volume_type           = var.root_vl_type
+    volume_size           = var.root_vl_size
+    delete_on_termination = var.root_vl_delete
   }
 
-  ebs_block_device = ["${var.ebs_block_devices}"]
+  dynamic "ebs_block_device" {
+    for_each = [var.ebs_block_devices]
+    content {
+      # TF-UPGRADE-TODO: The automatic upgrade tool can't predict
```

Comment:
Does this comment from tool need to stay?

Reply:
indeed the comment can go away. I think the actual implementation is ok